src/Tribe/Date_Utils.php

```diff
@@ -1289,13 +1301,26 @@ public static function is_valid_date( $date ) {
 		 *                        `23:59:59`.
 		 */
 		public static function get_week_start_end( $date, $start_of_week = null ) {
+			static $cache_var_name = __FUNCTION__;
+
+			$cache_week_start_end = tribe_get_var( $cache_var_name, [] );
+
 			$week_start = static::build_date_object( $date );
 			$week_start->setTime( 0, 0, 0 );
+
+			$date_string = $week_start->format( static::DBDATEFORMAT );
+
 			// `0` (for Sunday) through `6` (for Saturday), the way WP handles the `start_of_week` option.
 			$week_start_day = null !== $start_of_week
 				? (int) $start_of_week
 				: (int) get_option( 'start_of_week', 0 );
 
+			$memory_cache_key = "{$date_string}:{$week_start_day}";
+
+			if ( isset( $cache_week_start_end[ $memory_cache_key ] ) ) {
+				return $cache_week_start_end[ $memory_cache_key ];
+			}
+
 			$cache_key = md5(
 				__METHOD__ . serialize( [ $week_start->format( static::DBDATEFORMAT ), $week_start_day ] )
```

Comment:
Set this to static variable too, if we are doing it above for `__FUNCTION__`?

Reply:
Upon further digging, assigning `__FUNCTION__` and `__METHOD__` is likely more costly than just using the constant as is.

Reply:
A negligible cost, but not worth the micro-optimization of assigning to a static. (also not work the micro-optimization of rolling back the assignment)

Reply:
✅ just me asking the tough questions :)